src/Splits.sol

```diff
@@ -205,8 +218,16 @@ library Splits {
         SplitsReceiver[] memory receivers
     ) internal {
         assertSplitsValid(receivers);
-        s.splitsStates[userId].splitsHash = hashSplits(receivers);
-        emit SplitsUpdated(userId, receivers);
+        SplitsState storage state = s.splitsStates[userId];
+        bytes32 newSplitsHash = hashSplits(receivers);
+        if (newSplitsHash != state.splitsHash) {
+            state.splitsHash = newSplitsHash;
+            for (uint256 i = 0; i < receivers.length; i++) {
```

Comment:
Here it is the same we iterate them already in the `assertSplitsValid` function. Why not just emit the events there?

Reply:
Done, also we don't validate the configuration anymore if it's unchanged.